kratos/solving_strategies/strategies/residualbased_newton_raphson_strategy.h

```diff
@@ -259,6 +259,7 @@ class ResidualBasedNewtonRaphsonStrategy
      
     virtual ~ResidualBasedNewtonRaphsonStrategy()
     {
+        Clear();
```

Comment:
@msandre OK, the issue you mention is a good reason to force the clear. However, I think that having to call it here is indicative of confused ownership of stuff. I think it would make sense to store a copy of the mpA pointer in the solver/preconditioner, but I guess that if it was possible to do so without changing the interface you would have done so.

I'm not holding back the review for this (since it is beyond the scope of your fix), but we should keep it in mind going forward.

@RiccardoRossi  My reasoning for not calling Clear is that (ideally) it should do nothing: everything managed by this class should be correctly deallocated by the default destructor. Obviously this is not the case here, as @msandre noted.